codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/rustlang/RustModule.kt

```diff
@@ -5,23 +5,27 @@
 
 package software.amazon.smithy.rust.codegen.rustlang
 
-data class RustModule(val name: String, val rustMetadata: RustMetadata, val documentation: String?) {
+data class RustModule(val name: String, val rustMetadata: RustMetadata, val documentation: String?, val hidden: Boolean?) {
```

Comment:
Whether the module is doc hidden or not should be indicated via the input `RustMetadata`. Example:

```
RustMetadata(additionalAttributes = listOf(Attribute.DocHidden))
```

And then line 14 takes care of rendering the doc hidden attribute.